protocol-specs/generators/lisk_codec/types_generators/transaction.js

```diff
@@ -35,7 +35,7 @@ const baseTransactionSchema = {
 		asset: { dataType: 'bytes', fieldNumber: 6 },
 		signatures: { type: 'array', items: { dataType: 'bytes' }, fieldNumber: 7 },
 	},
-	required: ['type', 'nonce', 'fee', 'senderPublicKey', 'asset'],
+	required: ['moduleID', 'assetID', 'nonce', 'fee', 'senderPublicKey', 'asset', 'signatures'],
```

Comment:
why adding the requirement here?

Reply:
The actual error was, the `type` property was mentioned in `required` but its definition in `properties` were missing. Also noticed few of the properties which have defined `fieldNumber` are not mentioned in `required`. So its better the validation to fail than codec.

Reply:
probably not a problem in this case but it's possible to have property with fieldNumber which is not required